src/test/jtx/impl/flags.cpp

```diff
@@ -41,7 +41,9 @@ void
 flags::operator()(Env& env) const
 {
     auto const sle = env.le(account_);
-    if (sle->isFieldPresent(sfFlags))
+    if (!sle)
```

Comment:
Can we add a check inside the `flags(Account const& account, Args... args)` constructor, to check for null-pointer in `account_` data member? It is not useful to set flags on a non-existent, unfunded account.

However, I've only observed the `flags` object passed as an input into `env.require()`. If that continues, there's probably no operational difference between the `flags::constructor` and the `flags::operator()`

Reply:
No. In the `flags` ctor, `account_` is a value, not a pointer. It would be premature to check the ledger via an `Env` parameter because the ledger state may change in between the ctor and the call to `operator()`.

Reply:
That may be true, but why would you ever want to construct a `flags` object with an unfunded account? 
We can move the validity check up from the `operator()` into the constructor. 

I can't think of situations where you'd need to create a `flags(unfunded_account)`

Reply:
It's not that you would want to create a `flags(unfunded_account)`, it's that the way these helper classes are constructed, there's no way to check whether the account is funded from the constructor. 

Here's the definition of the ctor:
```
    template <class... Args>
    flags(Account const& account, Args... args)
        : flags_helper(args...), account_(account)
    {
    }
```

And here's the definition of the `operator()`:
```
void
flags::operator()(Env& env) const
{
    auto const sle = env.le(account_);
    if (!sle)
        env.test.fail();
    else if (sle->isFieldPresent(sfFlags))
        env.test.expect((sle->getFieldU32(sfFlags) & mask_) == mask_);
    else
        env.test.expect(mask_ == 0);
}
```

The difference is the `Env` parameter to `operator()`. There isn't one of those available to the constructor, _and we don't want one_, because it would defeat the whole purpose of these helper classes and the `env.require` helper wrapper.

In other words, to do the check in the ctor, the callers would have to look something like
```
env.require(flags(env, alice, flag));
```
Which is ugly and redundant.

The last problem, IMHO, with trying to do a check like this in the ctor is what do you do if it fails? You could do some variation of a `BEAST_EXPECT` that would cause the test suite to fail, but then what? You'd either have an object with a dud account inside it that's going to get `operator()` called on it anyway, or you're going to have to throw an exception, which is what I was trying to avoid with this check in the first place.

Reply:
okay, this is a helpful explanation.

Why do you want to avoid throwing an exception in the constructor? I vaguely remember reading about it, I think it has something to do with the cleanup of the dead object. If you know any resources/blogs, I'd like to read it.

Reply:
> Why do you want to avoid throwing an exception in the constructor?

There is nothing intrinsically wrong with throwing an exception in the ctor. It's a good way to handle certain types of errors where you want to prevent an object from being instantiated at all.

The reason I don't want to throw an exception in _this_ ctor is because exceptions need to be caught or they terminate the whole program. The good news is that in the test suite, the tests are all wrapped so that exceptions will be caught, and treated as a test suite error. The bad news is that code is a blunt instrument, which doesn't tell you much about where and why the exception occurred. It also causes the rest of that suite to be skipped, as opposed to a failing `BEAST_EXPECT`, which logs a failure including file and line number, and finishes the rest of the suite. That makes debugging a lot easier.

The alternative is to wrap _every single_ instantiation of `flags` in a `try/catch` block, which just adds unnecessary complication and syntactic garbage that doesn't add anything to the test.

So the short answer is that I don't want to throw in the ctor because it adds unnecessary complication when the check in `operator()` is perfectly sufficient.

Reply:
okay 👍  makes sense